File: adventofcode/year2018/Day01.cs
using adventofcode.common;

namespace adventofcode.year2018;

public class Day01 : Solution
{
    private List<int> _frequencies;
    
    public Day01(string year, string day) : base(year, day)
    {
        _frequencies = new List<int>();
        foreach (var line in LoadInputAsLines())
        {
            _frequencies.Add(Convert.ToInt32(line));
        }
    }

    public override string PartOne()
    {
        return Convert.ToString(_frequencies.Aggregate(0, (acc, f) => acc + f));
    }

    public override string PartTwo()
    {
        var seenFrequencies = new HashSet<int>();
        var frequency = 0;
        var repeat = 0;

        var index = 0;
        while (true)
        {
            frequency += _frequencies[index];
            if (seenFrequencies.Contains(frequency))
            {
                repeat = frequency;
                break;
            }
            seenFrequencies.Add(frequency);
            index = index + 1 >= _frequencies.Count ? 0 : index + 1;
        }
        
        return Convert.ToString(repeat);
    }
}

File: adventofcode/year2020/day01.py
from __future__ import annotations
from adventofcode.common import Solution
from itertools import combinations


class Day01(Solution):
    def __init__(self, year: str, day: str):
        super().__init__(year, day)
        self._input = [int(line) for line in self._load_input_as_lines()]

    def part_one(self):
        for a, b in combinations(self._input, 2):
            if a + b == 2020:
                return a * b
        return "ᕕ( ᐛ )ᕗ"

    def part_two(self):
        for a, b, c in combinations(self._input, 3):
            if a + b + c == 2020:
                return a * b * c
        return "ᕕ( ᐛ )ᕗ"
